tests/unit/s2n_wildcard_hostname_test.c

```diff
@@ -69,5 +68,177 @@ int main(int argc, char **argv)
         }
     }
 
+    /* s2n_connection_get_certificate_match */
+    {
+        /* Safety checks */
+        {
+            s2n_cert_sni_match match_status = 0;
+            struct s2n_connection *conn = NULL;
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_get_certificate_match(NULL, &match_status),
+                    S2N_ERR_INVALID_ARGUMENT);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_get_certificate_match(conn, NULL),
+                    S2N_ERR_INVALID_ARGUMENT);
+
+            /* This API does not work on a client connection */
+            DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(client_conn);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_get_certificate_match(client_conn, &match_status),
+                    S2N_ERR_CLIENT_MODE);
+
+            /* This API will not work if a certificate isn't selected yet. */
+            DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(server_conn);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_get_certificate_match(server_conn, &match_status),
+                    S2N_ERR_NO_CERT_FOUND);
+        }
+
+        /* Client does not send an SNI extension */
```

Comment:
Would it be neater to use tests cases here? Most of the logic looks to be the same between tests.
```C
struct SniTestCase {
    char* client_sni_indication,
    char* server_cert_key,
    char* server_cert_chain,
    s2n_cert_sni_match expected_match,
}
```

Reply:
yeah its a little cleaner